KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/LiveStreamSettingsModule.java

```diff
@@ -0,0 +1,300 @@
+				logger.info("adding new id3Frame [" + this.streamName + "] " + json);
+				id3Tag.setValue(json);
+				// it is essential to call clear. This method is used to indicate ID3 is set in specified chunk
+				// if clear() is not called multiple ID3 tag frame will be inserted to each chunk.
+				id3Frames.clear();
+				id3Frames.putFrame(id3Tag);
+
+			} catch (JsonProcessingException e) {
+				logger.error("stream [" + this.streamName + "] failed to add sync points data to ID3Tag " + e.toString());
+			}
+
+		}
+
+		public void onFillChunkEnd(LiveStreamPacketizerCupertinoChunk chunk, long timecode) {
+			logger.info("[" + chunk.getRendition().toString() + ":" + liveStreamPacketizer.getContextStr() + "]: chunkId:" + chunk.getChunkIndexForPlaylist());
+		}
+
+		public void onFillChunkMediaPacket(LiveStreamPacketizerCupertinoChunk chunk, CupertinoPacketHolder holder, AMFPacket packet) {
+			logger.info("[" + chunk.getRendition().toString() + ":" + liveStreamPacketizer.getContextStr() + "]: chunkId:" + chunk.getChunkIndexForPlaylist());
+		}
+
+		public void onFillChunkDataPacket(LiveStreamPacketizerCupertinoChunk chunk, CupertinoPacketHolder holder, AMFPacket packet, ID3Frames id3Frames) {
+			logger.info("[" + chunk.getRendition().toString() + ":" + liveStreamPacketizer.getContextStr() + "]: chunkId:" + chunk.getChunkIndexForPlaylist());
+		}
+
+	}
+
+	class LiveStreamPacketizerListener implements ILiveStreamPacketizerActionNotify {
+
+		private IApplicationInstance appInstance = null;
+
+		public LiveStreamPacketizerListener(IApplicationInstance appInstance) {
+			logger.debug("creating new LiveStreamPacketizerListener");
+			this.appInstance = appInstance;
+		}
+
+		public void onLiveStreamPacketizerCreate(ILiveStreamPacketizer liveStreamPacketizer, String streamName) {
+			logger.debug("onLiveStreamPacketizerCreate. stream: " + streamName);
+			if (!(liveStreamPacketizer instanceof LiveStreamPacketizerCupertino))
+				return;
+
+			LiveStreamPacketizerCupertino cupertinoPacketizer = (LiveStreamPacketizerCupertino) liveStreamPacketizer;
+			IMediaStream stream = this.appInstance.getStreams().getStream(streamName);
+
+			logger.info("Create [" + streamName + "]: " + liveStreamPacketizer.getClass().getSimpleName());
+			cupertinoPacketizer.setDataHandler(new LiveStreamPacketizerDataHandler(cupertinoPacketizer, streamName));
+		}
+
+		public void onLiveStreamPacketizerDestroy(ILiveStreamPacketizer liveStreamPacketizer) {
+			String streamName = liveStreamPacketizer.getAndSetStartStream(null).getName();
+			logger.debug("(" + streamName + ") onLiveStreamPacketizerDestroy");
+		}
+
+		public void onLiveStreamPacketizerInit(ILiveStreamPacketizer liveStreamPacketizer, String streamName) {
+			logger.debug("(" + streamName + ") onLiveStreamPacketizerInit");
+		}
+	}
+
+	public void onAppStart(IApplicationInstance applicationInstance) {
+		this.liveStreamPacketizerListener = new LiveStreamPacketizerListener(applicationInstance);
```

Comment:
don't know how to do it yet.
removeLiveStreamPacketizerListener(liveStreamPacketizerListener);